include/swift/AST/DiagnosticsParse.def

```diff
@@ -724,6 +724,9 @@ ERROR(rethrows_in_wrong_position,none,
       "'rethrows' may only occur before '->'", ())
 ERROR(throw_in_function_type,none,
       "expected throwing specifier; did you mean 'throws'?", ())
+ERROR(throw_in_function_type_and_wrong_position,none,
+      "expected throwing specifier; did you mean 'throws'? "
+      "'throws' may only occur before '->'", ())
```

Comment:
This should be in a separate note which applies a fix-it. You could then reuse the throw_in_function_type diagnostic. Also, it might be better to not have a separate not for this at all, but just reuse rethrows_in_wrong_position diagnostic. You can tweak it so it's %select{'throws'|'rethrows'}0 may only occur before '->' and then pass 1 when emitting this diagnostic.

Reply:
I would investigate this approach, reusing existing diagnostics sounds much better 👍 
I'm not sure how the notes system work yet.

Reply:
@theblixguy could you clarify what do you expect?

1. The error to only say `'throws' may only occur before '->'` but the fix it to apply both transformations?
2. The error to say `expected throwing specifier; did you mean 'throws'?` and an extra note with `'throws' may only occur before '->'`?
3. Something else? ¯\_(ツ)_/¯ 

I would like to clarify that I think it's important that the fix-it solves the 2 issues at the same time (position and renaming). That's why I think that spelling out both problems is important, but I maybe missing something. 

As a separate note, could you point me to an example where a note is attached to an error? I'm having hard time figure out how to create the `diagnose(throwsLoc, diag::throw_in_function_type)` and attach a note later, plus the fix it. 

Thanks!

Reply:
I mean 

1. Change `rethrows_in_wrong_position` to:

```
ERROR(rethrows_or_throws_in_wrong_position,none,
      "%select{'rethrows'|'throws'}0 may only occur before '->'", ())
```

2. Add a new note:

```
NOTE(throw_in_function_type_and_wrong_position_note,none,
     "'throws' may only occur before '->'", ())
``` 

3. Return `diagnose(throwsLoc, diag::throw_in_function_type_and_wrong_position, 1);`
4. Call `diagnose(throwsLoc, diag:: throw_in_function_type_and_wrong_position_note)` and then call `fixItInsert` & `fixItRemove` on it, instead of on `diagOpt`.